compiler/rustc_target/src/asm/powerpc.rs

```diff
@@ -0,0 +1,124 @@
+use super::{InlineAsmArch, InlineAsmType};
+use rustc_macros::HashStable_Generic;
+use std::fmt;
+
+def_reg_class! {
+    PowerPC PowerPCInlineAsmRegClass {
+        reg,
```

Comment:
PPC is a bit weird about r0 as well. Some instructions treat it as a literal zero, while others treat it as a register. How would we handle that?

Reply:
Use a separate register class to indicate "any register except `r0`". LLVM supports this with the `b` constraint code. Have a look at ARM which has `reg` and `reg_thumb` because some instructions only accept `r0` to `r7` in Thumb mode.

Reply:
It looks like adding a new register class is going to be more difficult than reusing an existing one.
Is there an appropriate existing class, or do you have an example changeset for adding a new class?

Reply:
I suggest looking at how ARM does it. Search for `reg_thumb` and just do the same thing.